hoomd/md/force.py

```diff
@@ -378,24 +377,45 @@ def _add(self, simulation):
         if simulation is not None:
             simulation._warn_if_seed_unset()
 
+        if self.manifold_constraint is not None:
+            self.manifold_constraint._add(simulation)
+
         super()._add(simulation)
 
     def _attach(self):
+
         # initialize the reflected c++ class
-        if isinstance(self._simulation.device, hoomd.device.CPU):
-            my_class = _md.ActiveForceCompute
+        sim = self._simulation
+
+        if self.manifold_constraint is None:
```

Comment:
Implement the manifold active force compute in a separate class to avoid issues like #998 for the integration methods.